pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -280,13 +280,18 @@ func (r *Reconciler) PushSecretToProviders(ctx context.Context, stores map[esapi
 			return out, fmt.Errorf("could not get secrets client for store %v: %w", store.GetName(), err)
 		}
 		for _, data := range ps.Spec.Data {
-			secretValue, ok := secret.Data[data.Match.SecretKey]
-			if !ok {
-				return out, fmt.Errorf("secret key %v does not exist", data.Match.SecretKey)
+			secretValue := map[string][]byte{}
+			if data.Match.SecretKey == "" {
+				secretValue = secret.Data
+			} else {
+				value, ok := secret.Data[data.Match.SecretKey]
+				if !ok {
+					return out, fmt.Errorf("secret key %v does not exist", data.Match.SecretKey)
+				}
+				secretValue[data.Match.SecretKey] = value
```

Comment:
Here we set the `secretValue[data.Match.SecretKey]`, below we use the `remoteRef.property` to look it up, which does not match.

We would have to pass the `data.Match.SecretKey` also into the provider so it is able to lookup the right key.

Reply:
I had a feeling that I'm not doing something correctly. :D :D Thanks! :)